common-rest/src/test/java/org/uniprot/api/rest/validation/ValidDownloadRequestTest.java

```diff
@@ -0,0 +1,64 @@
+package org.uniprot.api.rest.validation;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorContextImpl;
+import org.junit.jupiter.api.Test;
+import org.uniprot.api.rest.request.FakeDownloadRequest;
+
+class ValidDownloadRequestTest {
+
+    @Test
+    void when_fields_passed_then_invalid_request() {
+        FakeDownloadRequest downloadRequest = new FakeDownloadRequest();
+        downloadRequest.setFormat(ValidDownloadRequest.FORMATS_WITH_NO_PROJECTION.get(0));
+        downloadRequest.setFields("field1,field2");
+        FakeDownloadRequestValidator validator = new FakeDownloadRequestValidator();
+        boolean result = validator.isValid(downloadRequest, null);
+        assertFalse(result);
+        assertNotNull(validator.errorMessage);
+    }
+
+    @Test
+    void when_fields_null_then_valid_request() {
+        FakeDownloadRequest downloadRequest = new FakeDownloadRequest();
+        downloadRequest.setFormat(ValidDownloadRequest.FORMATS_WITH_NO_PROJECTION.get(0));
+        downloadRequest.setFields(null);
+        FakeDownloadRequestValidator validator = new FakeDownloadRequestValidator();
+        boolean result = validator.isValid(downloadRequest, null);
+        assertTrue(result);
+        assertNull(validator.errorMessage);
+    }
+
+    @Test
+    void when_fields_empty_then_valid_request() {
+        FakeDownloadRequest downloadRequest = new FakeDownloadRequest();
+        downloadRequest.setFormat(ValidDownloadRequest.FORMATS_WITH_NO_PROJECTION.get(0));
+        downloadRequest.setFields("");
+        FakeDownloadRequestValidator validator = new FakeDownloadRequestValidator();
+        boolean result = validator.isValid(downloadRequest, null);
+        assertTrue(result);
+        assertNull(validator.errorMessage);
+    }
+
+    @Test
+    void when_fields_passed_with_short_format_then_invalid_request() {
+        FakeDownloadRequest downloadRequest = new FakeDownloadRequest();
+        downloadRequest.setFormat("h5");
+        downloadRequest.setFields("field1,field2");
+        FakeDownloadRequestValidator validator = new FakeDownloadRequestValidator();
+        boolean result = validator.isValid(downloadRequest, null);
+        assertFalse(result);
+        assertNotNull(validator.errorMessage);
+    }
+
```

Comment:
Don't we need cases for when a valid format is passed?

Reply:
yes. i will add. thanks